numpy_inference.py

```diff
@@ -32,46 +34,77 @@ def main() -> None:
     test_images = mnist_test_data.get("test_images")
     test_labels = mnist_test_data.get("test_labels")
 
+    # Correct arguments if needed.
+    if args.batch_size > args.num_images:
+        args.num_images = args.batch_size
+
+    if args.num_images > len(test_images):
```

Comment:
This calculation should consider `start_image`, since that may reduce the effective number of available `test_images`